### ISSUE TITLE
Fantastic!

### ISSUE THREAD

**Issue by opensourcefan** [peterbaumert/netbox-device-view#1]
For an "alpha" this is excellent. Thank you for taking the time to build this.

**Comment by peterbaumert** [comment]
Thanks and please feel free to share your thoughts.

Still many things to do but i think too for a start this can already be better than nothing :)

**Comment by opensourcefan** [comment]
Yes, it's an excellent start, and a fast one to boot.  Really my only immediate thought was showing the active "front port" termination points. 

Curious if you'd be able to use the already set "front" images from the Device Types as an underlay, with your output being an overlay. We could adjust the x,y to match the image. Not sure if this is easier imagined then implementing.

**Comment by peterbaumert** [comment]
> Yes, it's an excellent start, and a fast one to boot. Really my only immediate thought was showing the active "front port" termination points.

Do you mean like this:

![image](https://github.com/peterbaumert/netbox-device-view/assets/1212934/c13309a2-ae88-4cdc-b024-f0194a005849)

**Comment by opensourcefan** [comment]
That would work. Was thinking maybe connections lines coming off at an angle top and bottom showing the connection. Something that would be visible without interaction and could be printed if needed. 


![switchlabel](https://github.com/peterbaumert/netbox-device-view/assets/10550038/86bc2e06-0911-4a8c-87c2-6abc20253f00)

**Comment by peterbaumert** [comment]
> That would work. Was thinking maybe connections lines coming off at an angle top and bottom showing the connection. Something that would be visible without interaction and could be printed if needed.
> 
> ![switchlabel](https://user-images.githubusercontent.com/10550038/243207000-bde95f31-7a11-4fec-8e9a-3958d365e944.png)

I think for the standard device info page, that might be a bit overloaded. But i added already a "Device View" tab to the devices page.

Now the bigger issue is that i am not very good at css, and i couldnt get a working "template" yet for this. If anybody is better at this, feel free to help :D

**Comment by PVM-NL** [comment]
Hi,

Thanks for the plugin was looking for this as i also created a issue for it back then https://github.com/netbox-community/netbox/issues/11931 , to me its almost the same as you are creating.

Wonder if you could add something like this?
As we use colorcode for switches to let people see if it is a interconnect or accesspoint position.

**Comment by peterbaumert** [comment]
> Hi,
> 
> 
> 
> Thanks for the plugin was looking for this as i also created a issue for it back then https://github.com/netbox-community/netbox/issues/11931 , to me its almost the same as you are creating.
> 
> 
> 
> Wonder if you could add something like this?
> 
> As we use colorcode for switches to let people see if it is a interconnect or accesspoint position.
> 
> 
> 
> 

Hey, I am currently in vacation till end of next week. But when i am back i will look into it. But i think this should be possible. Either with tags or maybe the cable color ( if you use this ). Or something completly different 😂

**Comment by peterbaumert** [comment]
> Hi,
> 
> Thanks for the plugin was looking for this as i also created a issue for it back then [netbox-community/netbox#11931](https://github.com/netbox-community/netbox/issues/11931) , to me its almost the same as you are creating.
> 
> Wonder if you could add something like this? As we use colorcode for switches to let people see if it is a interconnect or accesspoint position.

You mean like this e.g.

![image](https://github.com/peterbaumert/netbox-device-view/assets/1212934/74693304-1324-44ea-8fa6-01fecf01fd01)

**Comment by opensourcefan** [comment]
^^ that is really cool. Would it be possible to add a hover that reveals the connection?

**Comment by peterbaumert** [comment]
> > > Hi,
> > > Thanks for the plugin was looking for this as i also created a issue for it back then [netbox-community/netbox#11931](https://github.com/netbox-community/netbox/issues/11931) , to me its almost the same as you are creating.
> > > Wonder if you could add something like this? As we use colorcode for switches to let people see if it is a interconnect or accesspoint position.
> > 
> > 
> > You mean like this e.g.
> > ![image](https://user-images.githubusercontent.com/1212934/249124132-74693304-1324-44ea-8fa6-01fecf01fd01.png)
> 
> Awesome that almost the same yes, does this use the device roles for it? The legenda underneath with what the colors are could help in fast overview but thats extra.

Currently it uses the colors of the connected cables as configured.




> ^^ that is really cool. Would it be possible to add a hover that reveals the connection?

It is already, if you scroll little up there is a screenshot
